ProcessLib/GroundwaterFlow/CreateGroundwaterFlowProcess.cpp

```diff
@@ -23,6 +23,7 @@ std::unique_ptr<Process> createGroundwaterFlowProcess(
     Process::NonlinearSolver& nonlinear_solver,
     std::unique_ptr<Process::TimeDiscretization>&& time_discretization,
     std::unique_ptr<NumLib::ConvergenceCriterion>&& convergence_criterion,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
```

Comment:
The `createJacobianAssembler()` will create an analytical Jacobian assembler by default. If it is used, but the local assemblers of the specific process have insufficient capabilities, the default implementation of `LocalAssemblerInterface::assembleWithJacobian()` will make sure that OGS fails with a proper error message.